Add jedi_model_data to bundles

mranst opened the pull request:
I'm not exactly sure if this will work for #84, but I added the repo to bundles and added an option to not clone it if specified (again, not sure if this meets the mechanics of what you were looking for). I also threw in a change adding the date from the pinned versions to build.yaml so that can be consulted later

mranst committed:
add jedi_model_data to bundles

mranst committed:
add pinned version date to build.yaml

mranst committed:
Add option not to clone jedi_model_data

Dooruk commented:
Thanks for working on this. Did you try building a recent JEDI version using this branch?

mranst commented:
> Thanks for working on this. Did you try building a recent JEDI version using this branch?

I tested the cloning successfully, I wasn't sure if the build would work before the coordinated merge tomorrow, but I'll try it now

mranst committed:
pycodestyle fix

mranst committed:
Merge branch 'develop' into feature/jedi_model_data

mranst committed:
Merge branch 'develop' into feature/jedi_model_data

Dooruk commented:
@mranst thanks for working on this, can you test making a build now that the JEDI side is merged?

mranst commented:
@Dooruk I made a build with it yesterday, and ran 3dvar successfully

Dooruk reviewed (changes requested):
This looks good for me. I saw that you did a test installation.

1) Did you run ctests in the new build? I can't do it from your buld.
2) Also, sorry, I'm going to be demanding but would you be able to update the documentation as well? This part explains the configurations in the YAML and it would help users what these options mean and how did they become relevant (with the model-data change). It is not clear to me the purpose of `pinned_versions_date` if the pinned versions are for a different date does it automatically correct them?:

https://geos-esm.github.io/jedi_bundle/#/dynamic_configuration_options

mranst committed:
Update docs and remove pinned_versions_date

mranst committed:
Remove read_hash_date

mranst commented:
@Dooruk Which ctests are relevant? Some are failing but the majority are succeeding, my understanding it that's to be expected

Dooruk commented:
> @Dooruk Which ctests are relevant? Some are failing but the majority are succeeding, my understanding it that's to be expected

Is it a few that are failing?

mranst commented:
> Is it a few that are failing?

Yeah, that's how I'd describe it

mranst commented:
I added `download_lfs_data` to the docs. I removed the pinned versions date. My idea was that if you created build.yaml and somehow didn't know which date of pinned versions you used, you could look at the build.yaml, but that's probably unecessary

Dooruk commented:
Sounds good, I'm testing this on my end now too then I will merge.

Not sure if you followed the discussions on relevant PRs, because I didn't closely but do you know if repos like `ufo-data`, `ioda-data` and such are still necessaery?

mranst commented:
I don't know either, but I believe it only affected the saber and vader repos, so my guess is if they were necessary before they still are

Dooruk reviewed (approved):
All good, thanks 👍🏼  Only like 3-4 cases failed for me.
